cubewalkers/parser.py

```diff
@@ -169,3 +170,99 @@ def bnet2rawkernel(rules: str,
     cpp_body += '\n} else{A__reserved_output[a__reserved]=A__reserved_input[a__reserved];}}}'
 
     return cp.RawKernel(cpp_body, kernel_name), varnames, cpp_body
+
+def network_rules_from_cana(N: BooleanNetwork) -> str:
+    """Transforms the prime implicants LUT of a Boolean Network from CANA to algebraic format.
+
+    Parametershttps://github.com/rionbr/CANA
+    ----------
+    N : BooleanNetwork
+        CANA Boolean network
+
+    Returns
+    -------
+    str
+        Network rules in algebraic format.
+        Ex.: A* = A|B&C\nB* = C\nC* = A|B
+    """
+    
+    alg_rule = ""
+    int2name = {v: name_adjustment(k) for k, v in N.name2int.items()}
+    for node in N.nodes:
+        alg_rule += node_rule_from_cana(node=node, int2name=int2name)
+        alg_rule += "\n"
+    
+    return alg_rule[:-1]
+
+def node_rule_from_cana(node: BooleanNode, int2name: dict = None) -> str:

```

Comment:
`int2name: dict = None` should be `int2name: dict[int,str] | None = None`, the `dict` in the docstring should be `dict[int,str]`.

Reply:
Added this